### PR TITLE
Android emulator network addon

### PR DESCRIPTION
**Pull request by jlipps**
addon to adjust the network speed of an android emulator

### CODE REVIEW

**Comment by jlipps** [comment]
@mstrelex any objections to merging this?

File: android-network-speed/README.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# TestProject Addon - Android Network Speed
+
+This Addon provides an action that allows the setting of virtual network speed for Android (emulators only).
+
+## Author
+TestProject LTD. 2019
```
Comment (vitalybu) [inline android-network-speed/README.md]: Can probably be changed to 2020

File: android-network-speed/src/main/java/io/testproject/addon/android_network/AndroidNetworkAction.java [hunk 2]
```diff
@@ -28,17 +28,18 @@
...
-    @Parameter(description = "The network speed to adopt. Must be one of the following values: GSM, SCSD, GPRS, EDGE, UMTS, HSDPA, LTE, EVDO, FULL")
-    public String speed = "LTE";
+    @Parameter(description = "The network speed to adopt. Must be one of the following " +
+                             "values: GSM, SCSD, GPRS, EDGE, UMTS, HSDPA, LTE, EVDO, FULL")
+    public String speed = "";
 
     @Override
     public ExecutionResult execute(AndroidAddonHelper helper) throws FailureException {
         AndroidDriver<AndroidElement> driver = helper.getDriver();
         driver.setNetworkSpeed(NetworkSpeed.valueOf(speed));
```
Comment (vitalybu) [inline android-network-speed/src/main/java/io/testproject/addon/android_network/AndroidNetworkAction.java]: What about checking the validity `speed` string value?
As far as I can tell `NetworkSpeed.valueOf()` throws an `IllegalArgumentException` when passed a string that doesn't match any of the enum values, and this exception is not handled...

Comment (jlipps) [reply android-network-speed/src/main/java/io/testproject/addon/android_network/AndroidNetworkAction.java]: I figured that this exception would be handled by the overall framework. Is there some other way to flag to the framework that we encountered an error?

Comment (vitalybu) [reply android-network-speed/src/main/java/io/testproject/addon/android_network/AndroidNetworkAction.java]: You're probably right and the exception will be handled by the upper layer.
Was just wondering whether the exception message will be informative enough.
Let's wait for @mstrelex's input next week.

Comment (mstrelex) [reply android-network-speed/src/main/java/io/testproject/addon/android_network/AndroidNetworkAction.java:42]: Sorry it took me only half year, but yes- FailureException is handled on the upper layer.

**Pull request merged**